server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/GetAliasesRequest.java

```diff
@@ -33,6 +38,11 @@ public GetAliasesRequest(String... aliases) {
 
     public GetAliasesRequest() {}
 
+    /**
+     * NB prior to 8.12 get-aliases was a TransportMasterNodeReadAction so for BwC we must remain able to read these requests until we no
+     * longer need to support {@link org.elasticsearch.TransportVersions#CLUSTER_FEATURES_ADDED} and earlier. Once we remove this we can
```

Comment:
I can understand that BwC needs to be maintained. But could you please help me understand why `org.elasticsearch.TransportVersions#CLUSTER_FEATURES_ADDED` specifically? Is it because of the cluster state serialization changes? But I don't quite get how that is relevant here? It seems to me that it should be either the latest TransportVersions or some Feature constant?

Reply:
Really just that was the latest transport version when I started this branch, and there's no need to keep it up to date. We can move it to this attribute once https://github.com/elastic/elasticsearch/pull/101767 is merged.

Reply:
I opened https://github.com/elastic/elasticsearch/pull/102336 to address this.